### DISCUSSION TITLE
Auto-increment IDs

### DISCUSSION THREAD

**Discussion by retorquere** [blinkdb-js/blinkdb#33] [Q&A]
Is there a way to generate numeric auto-increment IDs rather than GUIDs as primary keys?

**Comment by maradotwebp** [comment]
You can write a helper function to do that.

```ts
// Initialize userTableId with the highest number not present already
let userTableId = Math.max(...(await many(userTable)).map(u => u.id + 1));
const genUserTableId = () => userTableId++;

// Now, whenever you create an entity, use `genUserTableId` to generate your ID.
await insert(userTable, { id: genUserTableId(), name: "Adrian", age: 24 });
```

**Reply by retorquere** [reply]
Might be faster to use Date.now() and check for existence.

**Reply by maradotwebp** [reply]
Also a solution :D ([users with stricter privacy settings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now?retiredLocale=de#reduced_time_precision) will cause more conflicts tho). As long as your ID is locally unique in a table, BlinkDB is fine with it.

**Reply by retorquere** [reply]
Didn't know that, but I think I'm not susceptible to it, my code runs in privileged (chrome) mode in Firefox.
